pkg/config/data.go

```diff
@@ -463,63 +390,43 @@ func (f *File) ValidConfig(verbose bool, out io.Writer) bool {
 }
 
 // Read decodes a disk file into an in-memory data structure.
-//
-// If reading from disk fails, then we'll use the static config embedded into
-// the CLI binary (which we expect to be valid). If an attempt to unmarshal
-// the static config fails then we have to consider something fundamental has
-// gone wrong and subsequently expect the caller to exit the program.
-//
-// NOTE: Some users have noticed that their profile data is deleted after
-// certain (nondeterministic) operations. The fallback to static config may be
-// contributing to this behaviour because if a read of the user's config fails
-// and we start to use the static version as a fallback, then that will contain
-// no profile data and that in-memory representation will be persisted back to
-// disk, causing the user's profile data to be lost.
-func (f *File) Read(path string, in io.Reader, out io.Writer, errLog fsterr.LogInterface) error {
-	var (
-		attempts int
-		data     []byte
-		retryErr error
-		readErr  error
-	)
-
-	// To help mitigate any loss of profile data within the user's config we will
-	// retry the file READ operation if it fails.
-	ctx := context.Background()
-	b := retry.NewConstant(1 * time.Second)
-	b = retry.WithMaxRetries(2, b) // attempts == 3 (first attempt + two retries)
-
-	retryErr = retry.Do(ctx, b, func(_ context.Context) error {
-		attempts++
-
-		// G304 (CWE-22): Potential file inclusion via variable.
-		// gosec flagged this:
-		// Disabling as we need to load the config.toml from the user's file system.
-		// This file is decoded into a predefined struct, any unrecognised fields are dropped.
-		/* #nosec */
-		data, readErr = os.ReadFile(path)
-		if readErr != nil {
-			errLog.Add(readErr)
+func (f *File) Read(
+	path string,
+	cfg []byte,
+	in io.Reader,
+	out io.Writer,
+	errLog fsterr.LogInterface,
+	verbose bool,
+) error {
+	f.static = cfg
```

Comment:
Not a blocker: it feels to me this the static configuration is no longer something that should be known outside here, so maybe internalise it and drop the parameter from Read?

Reply:
Done.